Sources/ProjectDescription/Target.swift

```diff
@@ -70,6 +70,12 @@ public struct Target: Codable, Equatable {
     /// Specifies whether if the target can be merged as part of another binary or not
     public var mergeable: Bool
 
+    /// The target's initial install tags associated with on demand resources
+    public var odrInitialInstallTags: [String]
+
+    /// The target's prefetched tag order associated with on demand resources
+    public var odrPrefetchOrder: [String]
```

Comment:
Both of these properties mirror build settings (`ON_DEMAND_RESOURCES_INITIAL_INSTALL_TAGS` and `ON_DEMAND_RESOURCES_PREFETCH_ORDER`) – why do we need a separate property for them?

What we need new API for is [managing tags](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/On_Demand_Resources_Guide/Tagging.html#//apple_ref/doc/uid/TP40015083-CH3-SW1). We can take inspiration from XcodeGen that has already added support for this: https://github.com/yonaskolb/XcodeGen/pull/753

I'd look into extending the `ResourceFileElement`, so that users can also specify a given tag for their resources. Before rewriting everything, let's first settle on the API, so you don't waste time on refactoring 🙏

Reply:
I agree that these two properties should be merged into one just to shorten the Target init signature. That's what I have already mentioned in one of my self-review comments:
https://github.com/tuist/tuist/pull/6178#discussion_r1563289807
I'm going to make these changes. 
As for the Xcodegen, I don't think they support tags for Asset Catalogs. 
I see that they had added support for file resources tags and that was already implemented in Tuist before me as a property of ResourceFileElement. 
But tagging Asset Catalogs resources is different. This kind of tags is a part of their Contents.json not a part of the Xcode project. If we ignore tags in Contents.json and make developers specify tags like it is done for file resources, this will conflict with Xcode approach.
Assets resources are accessible by their names, ie UIImage(named: "imageName"). 
File resources are accessible by their file urls in their bundle.
If we ignore Contents.json and go files resources way, we will break the standard accessors and, moreover, Tuist will become a required tool to keep a generated project working.
If we want to pass tags to ResourceFileElement everywhere and keep assets accessors, we will have to modify assets Contents.json which is not correct way in my opinion.
I tried to describe in detail how Xcode works with on-demand resources in the description of this PR and the goal was to translate this process clearly into Tuist code.